types/index.d.ts

```diff
@@ -7,15 +10,14 @@ declare namespace createError {
     statusCode?: number;
   }
 
-  export interface FastifyErrorConstructor {
-    new(a?: any, b?: any, c?: any): FastifyError;
-    (a?: any, b?: any, c?: any): FastifyError;
-    readonly prototype: FastifyError;
+  export interface FastifyErrorConstructor<E extends {} = {}> {
```

Comment:
`extends {} = {}` is afaik not typesafe.
Wont this be a potential issue? What about 
```suggestion
  export interface FastifyErrorConstructor<E extends { code: string; statusCode?: number; } = { code: string: statusCode?: number;}> {
```

Reply:
I tested it and the typings test run through. I think it is better to use my suggestion.

Reply:
Thank you for suggestion! I added that restriction.